Assets/MixedRealityToolkit/_Core/Devices/OpenVR/ViveWandController.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Devices;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Devices.OpenVR
+{
+    public class ViveWandController : GenericOpenVRController
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="trackingState"></param>
+        /// <param name="controllerHandedness"></param>
+        /// <param name="inputSource"></param>
+        /// <param name="interactions"></param>
+        public ViveWandController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+                : base(trackingState, controllerHandedness, inputSource, interactions) { }
+
+        //TODO - Update defaults
+        /// <summary>
+        /// The Generic OpenVR Controller default interactions.
+        /// </summary>
+        /// <remarks>A single interaction mapping works for both left and right controllers.</remarks>
+        public static readonly MixedRealityInteractionMapping[] DefaultViveWandInteractions =
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(2, "Grip Press",AxisType.Digital, DeviceInputType.GripPress, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(3, "Touchpad Position", AxisType.DualAxis, DeviceInputType.Touchpad, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(4, "Touchpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(5, "Touchpad Press", AxisType.Digital, DeviceInputType.TouchpadPress, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(6, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(7, "Thumbstick Press ", AxisType.Digital, DeviceInputType.ThumbPress, MixedRealityInputAction.None),
```

Comment:
vive wand doesn't have thumbsticks.

I was very explicit about what controller has what input in my PR.

```
+        public static readonly MixedRealityInteractionMapping[] DefaultLeftHandedInteractions =
+        {
+            // Controller Pose
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.PrimaryIndexTrigger
+            // Valve Knuckles Controller - Left Controller Trigger
+            new MixedRealityInteractionMapping(1, "Trigger Position", AxisType.SingleAxis, DeviceInputType.Trigger, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.PrimaryIndexTrigger
+            // Valve Knuckles Controller - Left Controller Trigger
+            new MixedRealityInteractionMapping(2, "Trigger Touch", AxisType.Digital, DeviceInputType.TriggerTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Axis1D.PrimaryIndexTrigger
+            new MixedRealityInteractionMapping(3, "Trigger Near Touch", AxisType.Digital, DeviceInputType.TriggerNearTouch, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.PrimaryIndexTrigger
+            // Valve Knuckles Controller - Left Controller Trigger
+            new MixedRealityInteractionMapping(4, "Trigger Press", AxisType.Digital, DeviceInputType.TriggerPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Grip Button (8)
+            // Oculus Touch Controller - Axis1D.PrimaryHandTrigger
+            // Valve Knuckles Controller - Left Controller Grip Average
+            new MixedRealityInteractionMapping(5, "Grip Average", AxisType.SingleAxis, DeviceInputType.TriggerPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Axis2D.PrimaryThumbstick
+            new MixedRealityInteractionMapping(6, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.PrimaryIndexTrigger
+            new MixedRealityInteractionMapping(7, "Thumbstick Touch", AxisType.Digital, DeviceInputType.ThumbStickTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.PrimaryIndexTrigger
+            new MixedRealityInteractionMapping(8, "Thumbstick Near Touch", AxisType.Digital, DeviceInputType.ThumbNearTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.PrimaryThumbstick
+            new MixedRealityInteractionMapping(9, "Thumbstick Press", AxisType.Digital, DeviceInputType.ThumbStickPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trackpad (2)
+            // Valve Knuckles Controller - Left Controller Trackpad
+            new MixedRealityInteractionMapping(10, "Trackpad Position", AxisType.DualAxis, DeviceInputType.Touchpad, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trackpad (2)
+            // Valve Knuckles Controller - Left Controller Trackpad
+            new MixedRealityInteractionMapping(11, "Trackpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Trackpad (2)
+            // Valve Knuckles Controller - Left Controller Trackpad
+            new MixedRealityInteractionMapping(12, "Trackpad Press", AxisType.Digital, DeviceInputType.TouchpadPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Left Controller Menu Button (1)
+            // Oculus Touch Controller - Button.Three Press
+            // Valve Knuckles Controller - Left Controller Inner Face Button
+            new MixedRealityInteractionMapping(13, "Unity Button Id 2", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.Four Press
+            // Valve Knuckles Controller - Left Controller Outer Face Button
+            new MixedRealityInteractionMapping(14, "Unity Button Id 3", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.Start Press
+            new MixedRealityInteractionMapping(15, "Unity Button Id 7", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.Three Touch
+            new MixedRealityInteractionMapping(16, "Unity Button Id 12", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.Four Touch
+            new MixedRealityInteractionMapping(17, "Unity Button Id 13", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Touch.PrimaryThumbRest Touch
+            new MixedRealityInteractionMapping(18, "Thumb Rest Touch", AxisType.Digital, DeviceInputType.ThumbTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Touch.PrimaryThumbRest Near Touch
+            new MixedRealityInteractionMapping(19, "Thumb Rest Near Touch", AxisType.Digital, DeviceInputType.ThumbNearTouch, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Left Controller Index Finger Cap Sensor
+            new MixedRealityInteractionMapping(20, "Index Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.IndexFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Left Controller Middle Finger Cap Sensor
+            new MixedRealityInteractionMapping(21, "Middle Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.MiddleFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Left Controller Ring Finger Cap Sensor
+            new MixedRealityInteractionMapping(22, "Ring Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.RingFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Left Controller Pinky Finger Cap Sensor
+            new MixedRealityInteractionMapping(23, "Pinky Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.PinkyFinger, MixedRealityInputAction.None),
+        };
+
+        public static readonly MixedRealityInteractionMapping[] DefaultRightHandedInteractions =
+        {
+            // Controller Pose
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.SecondaryIndexTrigger
+            // Valve Knuckles Controller - Right Controller Trigger
+            new MixedRealityInteractionMapping(1, "Trigger Position", AxisType.SingleAxis, DeviceInputType.Trigger, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.SecondaryIndexTrigger
+            // Valve Knuckles Controller - Right Controller Trigger
+            new MixedRealityInteractionMapping(2, "Trigger Touch", AxisType.Digital, DeviceInputType.TriggerTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Axis1D.SecondaryIndexTrigger
+            new MixedRealityInteractionMapping(3, "Trigger Near Touch", AxisType.Digital, DeviceInputType.TriggerNearTouch, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trigger (7)
+            // Oculus Touch Controller - Axis1D.SecondaryIndexTrigger
+            // Valve Knuckles Controller - Right Controller Trigger
+            new MixedRealityInteractionMapping(4, "Trigger Press", AxisType.Digital, DeviceInputType.TriggerPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Grip Button (8)
+            // Oculus Touch Controller - Axis1D.SecondaryHandTrigger
+            // Valve Knuckles Controller - Right Controller Grip Average
+            new MixedRealityInteractionMapping(5, "Grip Average", AxisType.SingleAxis, DeviceInputType.TriggerPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Axis2D.SecondaryThumbstick
+            new MixedRealityInteractionMapping(6, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.SecondaryIndexTrigger
+            new MixedRealityInteractionMapping(7, "Thumbstick Touch", AxisType.Digital, DeviceInputType.ThumbStickTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.SecondaryIndexTrigger
+            new MixedRealityInteractionMapping(8, "Thumbstick Near Touch", AxisType.Digital, DeviceInputType.ThumbNearTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.SecondaryThumbstick
+            new MixedRealityInteractionMapping(9, "Thumbstick Press", AxisType.Digital, DeviceInputType.ThumbStickPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trackpad (2)
+            // Valve Knuckles Controller - Right Controller Trackpad
+            new MixedRealityInteractionMapping(10, "Trackpad Position", AxisType.DualAxis, DeviceInputType.Touchpad, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trackpad (2)
+            // Valve Knuckles Controller - Right Controller Trackpad
+            new MixedRealityInteractionMapping(11, "Trackpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Trackpad (2)
+            // Valve Knuckles Controller - Right Controller Trackpad
+            new MixedRealityInteractionMapping(12, "Trackpad Press", AxisType.Digital, DeviceInputType.TouchpadPress, MixedRealityInputAction.None),
+            // HTC Vive Controller - Right Controller Menu Button (1)
+            // Oculus Remote - Button.One Press
+            // Oculus Touch Controller - Button.One Press
+            // Valve Knuckles Controller - Right Controller Inner Face Button
+            new MixedRealityInteractionMapping(13, "Unity Button Id 0", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Remote - Button.Two Press
+            // Oculus Touch Controller - Button.Two Press
+            // Valve Knuckles Controller - Right Controller Outer Face Button
+            new MixedRealityInteractionMapping(14, "Unity Button Id 1", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.One Touch
+            new MixedRealityInteractionMapping(15, "Unity Button Id 10", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Button.Two Touch
+            new MixedRealityInteractionMapping(16, "Unity Button Id 11", AxisType.Digital, DeviceInputType.ButtonPress, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Touch.SecondaryThumbRest Touch
+            new MixedRealityInteractionMapping(17, "Thumb Rest Touch", AxisType.Digital, DeviceInputType.ThumbTouch, MixedRealityInputAction.None),
+            // Oculus Touch Controller - Touch.SecondaryThumbRest Near Touch
+            new MixedRealityInteractionMapping(18, "Thumb Rest Near Touch", AxisType.Digital, DeviceInputType.ThumbNearTouch, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Right Controller Index Finger Cap Sensor
+            new MixedRealityInteractionMapping(19, "Index Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.IndexFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Right Controller Middle Finger Cap Sensor
+            new MixedRealityInteractionMapping(20, "Middle Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.MiddleFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Right Controller Ring Finger Cap Sensor
+            new MixedRealityInteractionMapping(21, "Ring Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.RingFinger, MixedRealityInputAction.None),
+            // Valve Knuckles Controller - Right Controller Pinky Finger Cap Sensor
+            new MixedRealityInteractionMapping(22, "Pinky Finger Cap Sensor", AxisType.SingleAxis, DeviceInputType.PinkyFinger, MixedRealityInputAction.None),
+        };
```